### ISSUE TITLE
Wrong grammar for negative number

### ISSUE THREAD

**Issue by dai-chen** [opendistro-for-elasticsearch/sql#488]
The negative integer and floating-point number is matched wrongly in current ANTLR grammar. This bug seems introduced since the very beginning when new ANTLR parser was added. Here is an example:

```
PUT float-test/_doc/1
{
  "balance": 1.23
}

PUT float-test/_doc/2
{
  "balance": 2.21
}

PUT float-test/_doc/3
{
  "balance": -746.0000000000075
}

POST _opendistro/_sql?pretty
{
  "query": """
    SELECT
      balance FROM null-sum-test WHERE balance > -746
  """
}
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "*Field [-746] cannot be found or used here.*",
    "type": "SemanticAnalysisException"
  },
  "status": 400
}

POST _opendistro/_sql?pretty
{
  "query": """
    SELECT
      balance FROM null-sum-test WHERE balance > -1.0
  """
}
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "Failed to parse query due to *offending symbol [.0]* at: 
    '\n    SELECT\n      balance FROM null-sum-test WHERE balance > -1.0' <--- HERE... 
    More details: extraneous input '.0' expecting <EOF>",
    "type": "SyntaxAnalysisException"
  },
  "status": 400
}
```

**Comment by bernony** [comment]
Hi everyone,
In my company, we use Open Distro to access ES. 
Yesterday, we realized negative float and integer values are not well handled by open distro. When we checked, we saw you already solved the issue(what is a good news) but not released yet.
As it  is critical for us and have no workaround, we'd like to know when gonna be the next release including that fix.

Thank you I advance.

BR

Bernony

**Comment by dai-chen** [comment]
Hi @bernony , the error was in our ANTLR parser grammar which is for semantic check. You can disable it for now and enable it to verify after new release. Thanks!

Reference for how to disable semantic analyzer: https://github.com/opendistro-for-elasticsearch/sql/blob/master/docs/user/admin/settings.rst#opendistro-sql-query-analysis-enabled

**Comment by bernony** [comment]
Hi @dai-chen . We tried it and it works. Thank you

**Comment by dai-chen** [comment]
> Hi @dai-chen . We tried it and it works. Thank you

Cool. Just to let you know, the fix is already included in upcoming 1.9 release. Thanks.
